### ISSUE TITLE
Uncaught TypeError: Cannot convert undefined or null to object in Document.invalidate

### ISSUE THREAD

**Issue by yanneves** [Automattic/mongoose#4706]
This error has cropped up out of the blue in my integration tests, can reproduce when trying to write any document using `Document.create()` with invalid data.

Using `mongoose@4.6.7` on Node v7.1.0

Full stack trace:
```
Uncaught TypeError: Cannot convert undefined or null to object
  at Function.keys (<anonymous>)
  at ValidationError.toString (node_modules\mongoose\lib\error\validation.js:51:10)
  at Function.prepareStackTrace (node_modules\source-map-support\source-map-support.js:303:16)
  at Function.captureStackTrace (<anonymous>)
  at new MongooseError (node_modules\mongoose\lib\error.js:12:11)
  at ValidationError (node_modules\mongoose\lib\error\validation.js:17:19)
  at model.Document.invalidate (node_modules\mongoose\lib\document.js:1480:32)
  at node_modules\mongoose\lib\document.js:1356:17
  at validate (node_modules\mongoose\lib\schematype.js:705:7)
  at node_modules\mongoose\lib\schematype.js:742:9
  at Array.forEach (native)
  at SchemaString.SchemaType.doValidate (node_modules\mongoose\lib\schematype.js:710:19)
  at node_modules\mongoose\lib\document.js:1354:9
  at _combinedTickCallback (internal/process/next_tick.js:67:7)
  at process._tickDomainCallback (internal/process/next_tick.js:122:9)
```

**Comment by Flibo** [comment]
I'm facing the exact same issue on `mongoose@4.6.7` on Node v7.0.0. This happens whenever I try to create any document with data that doesn't pass validation.

**Comment by vkarpov15** [comment]
Should be fixed with #4719
